contrib/storage-hive/core/src/main/java/org/apache/drill/exec/store/hive/schema/HiveDatabaseSchema.java

```diff
@@ -72,4 +80,76 @@ public String getTypeName() {
     return HiveStoragePluginConfig.NAME;
   }
 
+  @Override
+  public List<Pair<String, ? extends Table>> getTablesByNames(final List<String> tableNames) {
+    final String schemaName = getName();
+    final List<Pair<String, ? extends Table>> tableNameToTable = Lists.newArrayList();
+    List<org.apache.hadoop.hive.metastore.api.Table> tables;
+    // Retries once if the first call to fetch the metadata fails
+    synchronized(mClient) {
+      final List<String> tableNamesWithAuth = Lists.newArrayList();
+      for(String tableName : tableNames) {
+        try {
+          if(mClient.tableExists(schemaName, tableName)) {
```

Comment:
Here you are making a RPC call for every table. I thought for perf reasons we wanted to avoid the RPC call per table and instead use ```getTableObjectsByName``` to get all tables data in one RPC call. How does this patch improve the perf?